### DISCUSSION TITLE
About the ONNX Graph Static Optimization

### DISCUSSION THREAD

**Discussion by sgjzfzzf** [onnx/onnx#5898] [Q&A] [answered]
Hi, I'm new to ONNX and trying to write a simple compiler based on it. I wonder whether ONNX provides basic optimization passes like operators fusion and constant folds. I'm trying to find related work by reading the official document and googling it but I find nothing. I want a tool that could optimize an ONNX graph and save it as a new one, or I need to implement these passes myself. Thank you for reading this!

**Comment by andife** [comment, marked as answer]
Hi,
maybe you will have a look at 
https://github.com/daquexian/onnx-simplifier or https://github.com/onnx/optimizer

But there is also a SIG for Optimization: https://github.com/onnx/sigs/tree/main/optimizations

The optimization of the graph is hardware-dependent.... so you have to see what suits the target hardware.
Also https://github.com/openvinotoolkit/nncf might be an interesting tool for you?

**Comment by justinchuby** [comment]
We are developing tools for basic optimization and are expecting to release them soon. Like @andife suggested you may keep an eye on the development in the optimization SIG.

**Reply by justinchuby** [reply]
New optimization capabilities, under development, is provided with `onnxscript.optimizer`
